MarathonRecomp/api/Sonicteam/LuaNodeImp.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <Marathon.inl>
+#include <Sonicteam/LuaNode.h>
+#include <stdx/string.h>
+#include <boost/smart_ptr/shared_ptr.h>
+
+namespace Sonicteam
+{
+    class LuaNodeImp : public LuaNode
+    {
+    public:
+        struct Vftable : LuaNode::Vftable
+        {
+            be<uint32_t> fpGetTableString;
+            be<uint32_t> fpGetTableConstString;
+            be<uint32_t> fpGetTableConstStringFast;
+            be<uint32_t> fpGetRoot;
+            be<uint32_t> fpGetEmptyNode;
+            be<uint32_t> fpIsTableExist;
+            be<uint32_t> fpIsRoot;
+            be<uint32_t> fpSetTablePath;
+            be<uint32_t> fpCallFunction;
+            be<uint32_t> fpCallFunction2;
+            be<uint32_t> fpCallFunction3;
+            be<uint32_t> fpIsNumber;
+            be<uint32_t> fpGetNumber;
+            be<uint32_t> fpSetNumber;
+            be<uint32_t> fpGetNumberFloat;
+            be<uint32_t> fpSetNumberFloat;
+            be<uint32_t> fpGetString;
+            be<uint32_t> fpSetString;
+            be<uint32_t> fpIsBoolean;
+            be<uint32_t> fpGetBoolean;
+            be<uint32_t> fpSetBoolean;
+            be<uint32_t> fpSetNil;
+            be<uint32_t> fpRegisterFunctions;
+            be<uint32_t> fpFunc60;
+        };
+
+        boost::shared_ptr<LuaNodeImp> GetTable(stdx::string* tablePath)
+        {
+            auto result = guest_stack_var<boost::shared_ptr<LuaNodeImp>>();
+            auto vft = (Vftable*)m_pVftable.get();
+            GuestToHostFunction<void>(vft->fpGetTableString,result.get(), this, tablePath);
+            return *result.get();
+        }
+
+        void CallFunction3(const char* functionName, void* customData)
+        {
+            auto result = guest_stack_var<boost::shared_ptr<LuaNodeImp>>();
+            auto vft = (Vftable*)m_pVftable.get();
+            GuestToHostFunction<void>(vft->fpCallFunction3, this, functionName, customData);
+        }
+
+        void RegisterFunctions(lua50::luaL_reg* functions, uint32_t count)
+        {
+            auto result = guest_stack_var<boost::shared_ptr<LuaNodeImp>>();
+            auto vft = (Vftable*)m_pVftable.get();
+            GuestToHostFunction<void>(vft->fpRegisterFunctions, this, functions, count);
+        }
+
+        stdx::string m_LuaTable;
```

Comment:
Functions should always go below class members.

```suggestion
        stdx::string m_LuaTable;
        
        boost::shared_ptr<LuaNodeImp> GetTable(stdx::string* tablePath)
        {
            auto result = guest_stack_var<boost::shared_ptr<LuaNodeImp>>();
            auto vft = (Vftable*)m_pVftable.get();
            GuestToHostFunction<void>(vft->fpGetTableString, result.get(), this, tablePath);
            return *result.get();
        }

        void CallFunction3(const char* functionName, void* customData)
        {
            auto result = guest_stack_var<boost::shared_ptr<LuaNodeImp>>();
            auto vft = (Vftable*)m_pVftable.get();
            GuestToHostFunction<void>(vft->fpCallFunction3, this, functionName, customData);
        }

        void RegisterFunctions(lua50::luaL_reg* functions, uint32_t count)
        {
            auto result = guest_stack_var<boost::shared_ptr<LuaNodeImp>>();
            auto vft = (Vftable*)m_pVftable.get();
            GuestToHostFunction<void>(vft->fpRegisterFunctions, this, functions, count);
        }
```